subsys/power/device.c

```diff
@@ -41,10 +41,30 @@ static const char core_devices[NUM_CORE_DEVICES][MAX_DEV_NAME_LEN] = {
  * device power policies would be executed.
  */
 static int device_ordered_list[MAX_PM_DEVICES];
-static int device_retval[MAX_PM_DEVICES];
 static struct device *pm_device_list;
 static int device_count;
 
+#ifdef CONFIG_DEVICE_IDLE_PM
+int sys_pm_power_state_notify_devices(enum power_states state)
```

Comment:
@wentongwu Just FYI: there is a good app note on the power manager for TI SimpleLink devices you may wish to read: http://software-dl.ti.com/simplelink/esd/simplelink_cc13x2_26x2_sdk/3.20.00.68/exports/docs/tidrivers/Power_Management.pdf. There are some concepts in there (e.g. notifications, constraints, dependencies, resources) that you may find interesting. The source code is BSD licensed, and the ideas are generally applicable for other platforms.

Reply:
@vanti Thanks.